x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/engines/engines_list_logic.ts

```diff
@@ -61,6 +69,12 @@ interface EngineListValues {
   parameters: { meta: Meta; searchQuery?: string }; // Added this variable to store to the search Query value as well
   results: EnterpriseSearchEngine[]; // stores engine list value from data
   status: typeof FetchEnginesAPILogic.values.status;
+  isFetchEngineLoading: EngineViewValues['isLoadingEngine'];
+  isFetchEngineFlyoutVisible: boolean;
+  fetchEngineData: EngineViewValues['engineData']; // data from fetchEngineAPI
+  fetchEngineName: string | null;
+  fetchEngineApiError?: EngineViewValues['fetchEngineApiError'];
```

Comment:
I'm fine with connecting to the `EngineView` to re-use the actions/values from there since they work. But I think we should do it in a logic for the flyout instead of adding all of this to the engine list logic.

Reply:
I think creating a separate flyout logic for this stuff makes sense, to keep the logic file a little smaller and more easily readable.